### ISSUE TITLE
[BUG][CML]Headset audio stream has serious distortion and noise when hot plug headset during paplay music

### ISSUE THREAD

**Issue by YvonneYang2** [thesofproject/linux#993]
**Describe the bug**
Headset audio stream has serious distortion and noise when hot plug headset during paplay music. **Issue cannot reproduced with test recipe of ww22.1**

**Test recipe** 
Rander UCM https://github.com/RanderWang/intel-UCM/commits/hdmi
Firmware: b5bedb0 https://github.com/bardliao/sof/tree/dell-test
Kernel: 84361a7https://github.com/thesofproject/linux/tree/release/sof-v5.0 
Topology: file: tools/topology/sof-hda-generic.tplg, sof-hda-generic-2ch.tplg same as firmware
platform ：CML laptop 

**To Reproduce** 

1. Make sure Auto-Mute Mode in alsamixer is enable
2. paplay music
3. Hot plug headset during paplay music and check the audio in headset when headset is plugged in.

**Expected result**
Headset output streaming when headset is plugged in and internal speaker output streaming when headset is plugged out 

**Actual result**
Streaming output from headset has serious distortion and noise when headset is plugged in while audio from internal speaker is normal. 
Besides, only forced power off can recover the issue 

**Environment**
Rander UCM https://github.com/RanderWang/intel-UCM/commits/hdmi
Firmware: b5bedb0 https://github.com/bardliao/sof/tree/dell-test
Kernel: 0ce7065 https://github.com/thesofproject/linux/tree/release/sof-v5.0 
Topology: file: tools/topology/sof-hda-generic.tplg, sof-hda-generic-2ch.tplg same as firmware
platform ：CML laptop 

**dmesg**
```
[  116.572876] sof-audio-pci 0000:00:1f.3: pcm: open stream 0 dir 0
[  116.572882] sof-audio-pci 0000:00:1f.3: period min 192 max 16384 bytes
[  116.572885] sof-audio-pci 0000:00:1f.3: period count 2 max 16
[  116.572888] sof-audio-pci 0000:00:1f.3: buffer max 65536 bytes
[  116.573215] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  116.573394] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  116.573425] sof-audio-pci 0000:00:1f.3: format_val=49, rate=48000, ch=2, format=10
[  116.573440] sof-audio-pci 0000:00:1f.3: pcm: hw params stream 0 dir 0
[  116.573446] sof-audio-pci 0000:00:1f.3: generating page table for 000000005be213b1 size 0xffc0 pages 16
[  116.573477] sof-audio-pci 0000:00:1f.3: period_bytes:0x3fc0
[  116.573480] sof-audio-pci 0000:00:1f.3: periods:4
[  116.573502] sof-audio-pci 0000:00:1f.3: stream_tag 1
[  116.573518] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[  116.573693] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[  116.573697] sof-audio-pci 0000:00:1f.3: pcm: stream dir 0, posn mailbox offset is 790528
[  116.579808] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 0 dir 0 cmd 1
[  116.579822] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[  116.579997] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[  116.580007] sof-audio-pci 0000:00:1f.3: In hda_link_pcm_trigger cmd=1
[  116.835928] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  116.835962] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  117.335874] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  117.335907] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  122.731515] snd_hdac_bus_update_rirb: 244 callbacks suppressed
[  122.731538] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731552] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731559] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731565] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731571] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731578] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731584] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731590] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731597] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  122.731651] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  137.635783] snd_hdac_bus_update_rirb: 752 callbacks suppressed
[  137.635804] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635816] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635821] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635826] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635831] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635835] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635840] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635844] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635849] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  137.635897] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x3f0500
[  145.195826] snd_hdac_bus_update_rirb: 6309 callbacks suppressed
[  145.195847] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195861] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195867] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195873] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195878] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195884] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195889] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195895] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195900] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  145.195959] sof-audio-pci 0000:00:1f.3: spurious response 0x0:0x0, last cmd=0x2f0500
[  155.431995] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 0 dir 0 cmd 0
[  155.432015] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[  155.432248] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[  155.432265] sof-audio-pci 0000:00:1f.3: In hda_link_pcm_trigger cmd=0
[  155.432334] sof-audio-pci 0000:00:1f.3: pcm: free stream 0 dir 0
[  155.432340] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[  155.432549] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[  155.432586] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  155.432795] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  155.432923] sof-audio-pci 0000:00:1f.3: pcm: close stream 0 dir 0
[  156.335791] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  156.335817] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  157.499009] sof-audio-pci 0000:00:1f.3: ipc tx: 0x40010000: GLB_PM_MSG: CTX_SAVE
[  157.499230] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40010000: GLB_PM_MSG: CTX_SAVE
[  157.499275] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1010f0f successful
[  157.499292] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f successful
[  157.499299] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask f
[  157.499367] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000010 at  00000044
```

[paplay_noise_dmesg.log](https://github.com/thesofproject/linux/files/3235648/paplay_noise_dmesg.log)
[paplay_noise_logger.log](https://github.com/thesofproject/linux/files/3235649/paplay_noise_logger.log)

**Comment by wenqingfu** [comment]
fixed by #995?

**Comment by YvonneYang2** [comment]
Issue not reproduced with the test recipe below, so close it.

Rander UCM https://github.com/RanderWang/intel-UCM/commits/hdmi
Firmware: b5bedb0 https://github.com/bardliao/sof/tree/dell-test 
Kernel: 5a49a08 https://github.com/thesofproject/linux/tree/release/sof-v5.0 
Topology: file: tools/topology/sof-hda-generic.tplg, sof-hda-generic-2ch.tplg same as firmware
platform ：CML laptop
